gitpod-io/gitpod
# Prevent workspaces from being OOM killed

## Description
Prevent workspaces from being OOM killed. See the [RFC](https://www.notion.so/gitpod/Prevent-workspaces-from-being-OOM-killed-1ab8cd8b2d4d4f248c4dd8b0dc6d34f2) for details.

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/13254

## How to test
- Open workspace
- SSH into the VM
- Check the processes in the workspace (use /proc/[pid]/oom_score_adj). It should be -1000 for tier 1 processes and -100 for tier 2 processes

## Release Notes

```release-note
Ensure that workspace do not crash due to OOM kills
```

## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
